### PR TITLE
Add more logging to attendance command

### PR DESCRIPTION
**Pull request by H-Shay**
As the title states. Part of the effort to add some more useful logging to the commands.

### CODE REVIEW

File: src/commands/AttendanceCommand.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ limitations under the License.
 */
 
 import { ICommand } from "./ICommand";
-import { MatrixClient } from "matrix-bot-sdk";
```
Comment (H-Shay) [inline src/commands/AttendanceCommand.ts:18]: This is a drive-by cleanup - the import was unused.

File: src/commands/AttendanceCommand.ts [hunk 2]
```diff
@@ -90,12 +89,22 @@ export class AttendanceCommand implements ICommand {
...
         }
         for (const spiRoom of this.conference.storedInterestRooms) {
             const doAppend = !!targetAudId && (targetAudId === "all" || targetAudId === await spiRoom.getId());
             const inviteTargets = await this.conference.getInviteTargetsForInterest(spiRoom);
-            await append(inviteTargets, null, await spiRoom.getId(), spiRoom.roomId, null, doAppend);
+            try {
+                await append(inviteTargets, null, await spiRoom.getId(), spiRoom.roomId, null, doAppend);
+            }
+            catch (error) {
+                throw new Error(`Error calculating invite acceptance in special interest room ${spiRoom}: ${error.toString()}`)
```
Comment (Half-Shot) [inline src/commands/AttendanceCommand.ts:106]:
```suggestion
                throw new Error(`Error calculating invite acceptance in special interest room ${spiRoom}`, {cause: error });
```

Gives you a nice stacktrack

```
> new Error('Oh no, an error', { cause: new Error('caused by this cheeky thing')})
Error: Oh no, an error
    at REPL2:1:1
    at Script.runInThisContext (node:vm:121:12)
    ... 7 lines matching cause stack trace ...
    at [_line] [as _line] (node:internal/readline/interface:887:18) {
  [cause]: Error: caused by this cheeky thing
      at REPL2:1:39
      at Script.runInThisContext (node:vm:121:12)
      at REPLServer.defaultEval (node:repl:593:29)
      at bound (node:domain:432:15)
      at REPLServer.runBound [as eval] (node:domain:443:12)
      at REPLServer.onLine (node:repl:923:10)
      at REPLServer.emit (node:events:526:35)
      at REPLServer.emit (node:domain:488:12)
      at [_onLine] [as _onLine] (node:internal/readline/interface:416:12)
      at [_line] [as _line] (node:internal/readline/interface:887:18)
}
```

This feature is not used nearly enough https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#cause

File: src/commands/AttendanceCommand.ts [hunk 3]
```diff
@@ -90,12 +89,22 @@ export class AttendanceCommand implements ICommand {
             const bs = this.conference.getAuditoriumBackstage(await auditorium.getId());
             const inviteTargets = await this.conference.getInviteTargetsForAuditorium(auditorium);
             const bsInviteTargets = await this.conference.getInviteTargetsForAuditorium(auditorium, true);
-            await append(inviteTargets, bsInviteTargets, await auditorium.getId(), auditorium.roomId, bs.roomId, doAppend);
+            try {
+                await append(inviteTargets, bsInviteTargets, await auditorium.getId(), auditorium.roomId, bs.roomId, doAppend);
+            }
+            catch (error) {
+                throw new Error(`Error calculating invite acceptance in auditorium ${auditorium}: ${error.toString()}`)
```
Comment (Half-Shot) [inline src/commands/AttendanceCommand.ts:96]:
```suggestion
                throw new Error(`Error calculating invite acceptance in auditorium ${auditorium}`, { cause: error });
```

**Review by Half-Shot — APPROVED** [review]
Looks good. I've suggested a way to keep the stacktraces :)

File: src/commands/AttendanceCommand.ts (hunk 2 above)
Comment (H-Shay) [reply src/commands/AttendanceCommand.ts]: Cool thanks for the insight - I am new to typescript/javascript so appreciate the tips :)

**Comment by H-Shay** [comment]
Merged #195 into main.
